lib/stream.js

```diff
@@ -136,6 +136,39 @@ module.exports = class UDXStream extends streamx.Duplex {
     this.emit('connect')
   }
 
+  changeRemote (socket, remoteId, port, host) {
+    if (this._remoteChanging) throw new Error('Remote already changing')
+
+    if (!this.connected) throw new Error('Not yet connected')
+    if (socket.closing) throw new Error('Socket is closed')
+
+    if (this.socket.udx !== socket.udx) {
+      throw new Error('Cannot change to a socket on another UDX instance')
+    }
+
+    if (!host) host = '127.0.0.1'
+
+    const family = ip.isIP(host)
+    if (!family) throw new Error(`${host} is not a valid IP address`)
+    if (!(port > 0 && port < 65536)) throw new Error(`${port} is not a valid port`)
+
+    if (this.socket !== socket) this._previousSocket = this.socket
+
+    this.remoteId = remoteId
+    this.remotePort = port
+    this.remoteHost = host
+    this.remoteFamily = family
+    this.socket = socket
+
+    this._remoteChanging = once(this, 'remote-changed')
```

Comment:
if the stream is destroyed while waiting for this, does this deadlock?

Reply:
That's actually a good question 🤔 @jthomas43 Do you know if the callback will always fire, even if a destroy packet is queued on the new socket?